src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/ExportClientResponse.cs

```diff
@@ -14,7 +12,6 @@ protected ExportClientResponse(bool success, DateTime deadlineUtc, Exception? ex
         this.DeadlineUtc = deadlineUtc;
     }
 
-    [MemberNotNullWhen(false, nameof(Exception))]
```

Comment:
Related to my other comment: https://github.com/open-telemetry/opentelemetry-dotnet/pull/6000/files#r1857224013

Seems this line was trying to establish a contract that Exception is not null when Success is false.